### PR TITLE
973-intercom-support

### LINKED ISSUE #973: Intercom support

### PR DESCRIPTION
**Pull request by mainawycliffe**
Closes #973

### CODE REVIEW

File: src/components/Intercom/SetupIntercom.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  const { user } = useUser();
+
+  useEffect(() => {
+    if (typeof window === "undefined") {
+      return;
+    }
+
+    window.Intercom("boot", {
+      app_id: process.env.NEXT_PUBLIC_INTERCOM_APP_ID,
+      email: user?.email,
```
Comment (moshloop) [inline src/components/Intercom/SetupIntercom.tsx:18]: Need the name, clerk org name, id and clerk user id

File: src/components/Intercom/intercomChecklist.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  "LinkedConfigToIncident" = "LinkedConfigToIncident",
+  "LinkedLogToIncident" = "LinkedLogToIncident",
+  "LinkedHealthCheckToIncident" = "AddHealthCheckToIncident",
+  "CreatedHealthCheck" = "CreatedHealthCheck",
+  "CreatedConfigScrapper" = "CreatedConfigScrapper",
+  "CreatedLogBackend" = "CreatedLogBackend"
+}
+
+// Investigate if we can use this to track the user
+export function sendIntercomChecklistEvent(
```
Comment (moshloop) [inline src/components/Intercom/intercomChecklist.ts:17]: Move this to services/analytics.js and call the function, sendAnalyticEvent - we don't wont intercom spread all over the app

+ if no intercom appi id specified this should be a no-op

File: src/components/Intercom/SetupIntercom.tsx (hunk 1 above)
Comment (mainawycliffe) [reply src/components/Intercom/SetupIntercom.tsx]: Organization ID and Name might need to be an array, since a user can have multiple and we don't want to override them when they change organization

File: src/components/Intercom/intercomChecklist.ts (hunk 2 above)
Comment (mainawycliffe) [reply src/components/Intercom/intercomChecklist.ts]: The window.Intercom function should only be available when the intercom has been initialized, which only happens when the APP_ID is available.

File: src/api/query-hooks/mutations/useSettingsResourcesMutations.tsx [hunk 3]
```diff
@@ -27,6 +31,9 @@ export const useSettingsDeleteResource = (
     {
       onSuccess: () => {
         // force refetch of all resources
+        sendAnalyticEvent(resourceInfo.analyticsEventName.Deleted, {
+          resource: resourceInfo.name
+        });
```
Comment (moshloop) [inline src/api/query-hooks/mutations/useSettingsResourcesMutations.tsx:36]: It probably better to just:  sendResourceDeletedEvent(resourceInfo.name)

File: src/components/SchemaResourcePage/resourceTypes.tsx [hunk 4]
```diff
@@ -39,13 +40,21 @@ export type SchemaResourceType = {
     default?: any;
     hidden?: boolean;
   }[];
+  analyticsEventName: {
```
[the remark this answers is not in the document]
Comment (mainawycliffe) [reply src/components/SchemaResourcePage/resourceTypes.tsx:43]: @moshloop Because of how the UI is setup, I found it easier to embedded a resource events to the resources list, so when you delete, create or update, we are able to send the appropriate event.

**Comment by moshloop** [comment]
@mainawycliffe  please see comments + can we not add a default intercom id - it should be specifed as an environment variable and be a no-op if none specified

**Comment by mainawycliffe** [comment]
@moshloop remove the app from env.local, just left a comment for future guidance. We don't setup intercom in cases where app id is unavailable. Add the APP_ID env variable to the clerk based backend only, for now.

File: src/components/IncidentDetails/AddResponders/AddResponderModal.tsx [hunk 5]
```diff
@@ -193,6 +197,7 @@ export default function AddResponderModal({
       responder: Omit<Responder, "id" | "updated_at" | "created_at">
     ) => saveResponder(responder),
     onSuccess: (_) => {
+      sendAnalyticEvent(AnalyticEventsName.OpenedCreateIncidentModal);
```
Comment (moshloop) [inline src/components/IncidentDetails/AddResponders/AddResponderModal.tsx:200]: Wrong event

File: src/components/IncidentDetails/AddDefinitionOfDone/useAddCommentAsDoD.tsx [hunk 6]
```diff
@@ -14,6 +18,7 @@ export default function useAddCommentAsDoD(
   const { isLoading, mutate } = useCreateEvidenceMutation(
     {
       onSuccess: (evidence) => {
+        sendAnalyticEvent(AnalyticEventsName.OpenedCreateIncidentModal);
```
Comment (moshloop) [inline src/components/IncidentDetails/AddDefinitionOfDone/useAddCommentAsDoD.tsx:21]: Should not send events for just opening a modal

File: src/components/LogBackends/LogBackendsForm.tsx [hunk 7]
```diff
@@ -1,40 +1,42 @@
+import { useQueryClient } from "@tanstack/react-query";
```
Comment (moshloop) [inline src/components/LogBackends/LogBackendsForm.tsx:1]: Import optimisation and refactoring must be in their own commit

File: src/services/analytics.ts [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+  DeletedSettingLogsBackend = "DeletedSettingLogsBackend",
+
+  // Connections
+  CreatedSettingConnection = "CreatedSettingConnection",
+  UpdatedSettingConnection = "UpdatedSettingConnection",
+  DeletedSettingConnection = "DeletedSettingConnection",
+  // Properties
+  CreatedSettingProperty = "CreatedSettingProperty",
+  UpdatedSettingProperty = "UpdatedSettingProperty",
+  DeletedSettingProperty = "DeletedSettingProperty"
```
Comment (moshloop) [inline src/services/analytics.ts:54]: Don't need these events

File: src/services/analytics.ts [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+  "OpenedCreateIncidentModal" = "OpenedCreateIncidentModal",
+  "AddedResponderToIncident" = "AddedResponderToIncident",
+  "AddedDoDToIncident" = "AddedDoDToIncident",
+  "LinkedConfigToIncident" = "LinkedConfigToIncident",
+  "LinkedLogToIncident" = "LinkedLogToIncident",
+  "LinkedHealthCheckToIncident" = "AddHealthCheckToIncident",
+  "CreatedHealthCheck" = "CreatedHealthCheck",
+  "CreatedConfigScrapper" = "CreatedConfigScrapper",
+  "CreatedLogBackend" = "CreatedLogBackend",
+  "ViewedPage" = "ViewedPage",
```
Comment (moshloop) [inline src/services/analytics.ts:18]: Sort this list,  + rename AnalyticEventsName to just Events
